tests/rados/test_osd_compaction.py

```diff
@@ -0,0 +1,207 @@
+            pool=pool_name,
+            obj=obj_rand,
+            ops="setomapval",
+            key=custom_key,
+            val=custom_val,
+        )
+        time.sleep(5)
+
+        # verifying addition of custom OMAP key
+        # 4. Fetch value of a particular omap key
+        log.info(f"#4 Fetching value of a particular OMAP key obj: {obj_rand}")
+        omap_val = pool_obj.perform_omap_operation(
+            pool=pool_name, obj=obj_rand, ops="getomapval", key=custom_key
+        )
+        assert omap_val == custom_val
+        log.info(f"Value of OMAP key {custom_key} is: {omap_val}")
+
+        # 5. Remove the newly added OMAP entry
+        log.info(f"#5 Removing a custom OMAP entry for obj: {obj_rand}")
+        pool_obj.perform_omap_operation(
+            pool=pool_name, obj=obj_rand, ops="rmomapkey", key=custom_key
+        )
+        time.sleep(5)
+
+        # verifying successful removal of custom OMAP key
+        out = pool_obj.perform_omap_operation(
+            pool=pool_name, obj=obj_rand, ops="getomapval", key=custom_key
+        )
+        log.info(f"Output: {out}")
+        assert "No such key" in out
+
+        # 6. Clear all OMAP entries from an object
+        obj_rand = random.choice(omap_obj_list)
+        omap_obj_list.remove(obj_rand)
+        log.info(f"#6 Clearing all OMAP entries for obj: {obj_rand}")
+        pool_obj.perform_omap_operation(pool=pool_name, obj=obj_rand, ops="clearomap")
+        time.sleep(5)
+
+        # verifying removal of all omap entries for the obj
+        omap_keys = pool_obj.perform_omap_operation(
+            pool=pool_name, obj=obj_rand, ops="listomapkeys"
+        )
+        assert omap_keys is None
+        log.info(f"All OMAP entries succesfully removed for obj: {obj_rand}")
+
+        log.info("OMAP operations have been successfully executed")
+
+        # Fetch an Object with OMAPs and its primary OSD to perform OSD compaction
+        obj_compact = random.choice(omap_obj_list)
+        osd_id = rados_obj.get_osd_map(pool=pool_name, obj=obj_compact)[
+            "acting_primary"
+        ]
+        log.info(
+            f"Primary OSD of the object {obj_compact} to be used for compaction: {osd_id}"
+        )
+
+        # create another pool for background IOPS
+        bench_pool_name = bench_config["pool_name"]
+        assert rados_obj.create_pool(pool_name=bench_pool_name, **bench_config)
+
+        # Initiating background IOPS using rados bench for 5 mins
+        log.info("Initiating background IOPS using rados bench for 5 mins")
+        rados_obj.bench_write(
+            pool_name=bench_pool_name, rados_write_duration=300, background=True
+        )
+
+        # with IOPS in-progress perform OSD compaction of chosen OSD
+        log.info(f"Starting OSD compaction for OSD {osd_id}")
+        out, _ = cephadm.shell([f"ceph tell osd.{osd_id} compact"])
+        log.info(out)
+        assert "elapsed_time" in out
+        time.sleep(10)
```

Comment:
Later you can add one more check, If IOPS is not running then after the compaction the available memory of OSD should be greater than or equal to the previous memory.

Reply:
Added this check before running IOPS

Reply:
`Starting OSD compaction for OSD 13
2023-11-16 15:10:44,965 (cephci.test_osd_compaction) [INFO] - cephci.forked-repo-osd-compact.cephci.ceph.ceph.py:1557 - Running command cephadm -v shell -- ceph tell osd.13 compact on 10.0.208.53 timeout 600
2023-11-16 15:10:47,527 (cephci.test_osd_compaction) [INFO] - cephci.forked-repo-osd-compact.cephci.ceph.ceph.py:1591 - Command completed successfully
2023-11-16 15:10:47,529 (cephci.test_osd_compaction) [INFO] - cephci.forked-repo-osd-compact.cephci.tests.rados.test_osd_compaction.py:185 - {
    "elapsed_time": 0.33700545399999998
}

2023-11-16 15:10:57,534 (cephci.test_osd_compaction) [INFO] - cephci.forked-repo-osd-compact.cephci.ceph.ceph.py:1557 - Running command cephadm -v shell -- ceph osd df osd.13 -f json on 10.0.208.53 timeout 300
2023-11-16 15:10:59,659 (cephci.test_osd_compaction) [INFO] - cephci.forked-repo-osd-compact.cephci.ceph.ceph.py:1591 - Command completed successfully
2023-11-16 15:10:59,661 (cephci.test_osd_compaction) [INFO] - cephci.forked-repo-osd-compact.cephci.tests.rados.test_osd_compaction.py:194 - **Initial OSD utilization before compaction: 1.5901074630801688**
2023-11-16 15:10:59,662 (cephci.test_osd_compaction) [INFO] - cephci.forked-repo-osd-compact.cephci.tests.rados.test_osd_compaction.py:195 - **Final OSD utilization after compaction: 1.2836630967729332**`